model/transformer.py

```diff
@@ -0,0 +1,124 @@
+"""
+  Model Class for Transformer
+"""
+
+import math
+import torch
+import torch.nn as nn
+
+from torch.nn import Transformer
+from yacs.config import CfgNode
+
+from data_pipeline.dataset import BOS_IDX
+
+class PositionalEncoding(nn.Module):
+  """
+  Positional encoding for transformer model
+  """
+  def __init__(self,
+               d_model: int,
+               dropout=0.1,
+               max_len=500):
+    super(PositionalEncoding, self).__init__()
+    self.dropout = nn.Dropout(p=dropout)
+
+    positional_encoding = torch.zeros(max_len, d_model)
+    position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
+    div_term = torch.exp(torch.arange(
+        0, d_model, 2).float() * (-math.log(10000.0) / d_model))
```

Comment:
what does 10000 represent?

Reply:
it is used to compute the wavelength for the sine/cosine; taken from formula in _"Attention is All You Need"_
PE(pos, 2i) = sin(pos/10000^(2i/dmodel))
PE(pos, 2i+1) = cos(pos/10000^(2i/dmodel))